### DISCUSSION TITLE
issues trying to minify html but not process it further

### DISCUSSION THREAD

**Discussion by danieltroger** [parcel-bundler/parcel#5889] [Q&A] [answered]
Minifying css with bundle-text and the entry in .parcelrc works great, so I thought I'd attempt the same for HTML.

I tried including things with `bundle-text:./file.html` but I have multiple "issues":

Parcel says it needs @parcel/packager-html but it isn't satisfied with nightly, it wants a specific nightly:
```
🚨 Build failed.
@parcel/package-manager: Could not find module "@parcel/packager-html" satisfying 2.0.0-nightly.600+f28eaf60.
[project_directory]/package.json:54:5
  53 |     "@parcel/packager-css": "nightly",
> 54 |     "@parcel/packager-html": "nightly",
>    |     ^^^^^^^^^^^^^^^^^^^^^^^ Found this conflicting local requirement.
  55 |     "@parcel/transformer-css": "nightly",
  56 |     "@parcel/transformer-html": "nightly",
```

Well, at least it looks like it. But I think I just forgot to run `npm install`. Maybe parcel should check what actually is installed?

```
🚨 Build failed.
@parcel/package-manager: Could not find module "@parcel/packager-html" satisfying 2.0.0-nightly.600+f28eaf60.
[project_directory]/package.json:54:5
  53 |     "@parcel/packager-css": "nightly",
> 54 |     "@parcel/packager-html": "2.0.0-nightly.600+f28eaf60",
>    |     ^^^^^^^^^^^^^^^^^^^^^^^ Found this conflicting local requirement.
  55 |     "@parcel/transformer-css": "nightly",
  56 |     "@parcel/transformer-html": "nightly",
```

Next question:

How can I disable that parcel tries to find all my images? I include template-string-like stuff and dynamically replace it with regex in javascript, but parcel doesn't let me build it:

```
@parcel/core: Failed to resolve '${image_url}' from './lib/templates/file.html'
```

And last question: if I remove all my images, the output is still not minified. Does #5755 also exist for html? How can I _just minify_?

**Comment by mischnic** [comment, marked as answer]
> How can I disable that parcel tries to find all my images?

Your situation sounds similar to this: https://github.com/parcel-bundler/parcel/discussions/5863

> How can I just minify?

Maybe something like this works:
```
{
  "extends": "@parcel/config-default", 
  "transformers": {
    "bundle-text:*.html": ["@parcel/transformer-inline-string"],
  },
  "optimizers": {
    "bundle-text:*.html: ["@parcel/optimizer-htmlnano"]
  }
}
```

(So removing the `"..."`  from the transformer pipeline definiton in the default config)

> Does #5755 also exist for html

Yes, it's a fundamental problem, unrelated to the file extension but rather to `bundle-text:`.

**Reply by danieltroger** [reply]
> Your situation sounds similar to this: #5863

Nice, thanks for the link.
With your .terserrc however it seems to work without a custom resolver?

> Maybe something like this works:

HTML minification didn't work at first but I figured out I didn't have `@parcel/babel-plugin-transform-runtime` and `@parcel/babel-preset-env` on `nightly` and now it works flawlessly! Both css and html get minified and it lets me use my template variables.

THANK YOU a lot. This is so cool! Parcel is basically a unicorn now for what I need 😁

Btw what do you think about creating a "downloadable example repos" page in the docs to get newcomers started quicker? Parcel does exactly what I need now and it would be great to get others there quickly.

> Yes, it's a fundamental problem, unrelated to the file extension but rather to bundle-text:.

Ah, I see.

**Reply by mischnic** [reply]
> it seems to work without a custom resolver?

Right, I just read your comment top to bottom 😄 

> Btw what do you think about creating a "downloadable example repos" page in the docs to get newcomers started quicker? Parcel does exactly what I need now and it would be great to get other's there quickly.

The docs aren't final yet (and we're not happy with them yet). There's also https://github.com/parcel-bundler/examples, but that wasn't updated at all since Parcel 2.
